driver-core/src/test/functional/com/mongodb/client/model/expressions/DocumentExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,235 @@
+    private final DocumentExpression ax1ay2 = ofDoc("{a: {x: 1, y: 2}}");
+
+    @Test
+    public void literalsTest() {
+        assertExpression(
+                BsonDocument.parse("{'a': 1}"),
+                ofDoc("{a: 1}"),
+                "{'$literal': {'a': 1}}");
+        assertThrows(IllegalArgumentException.class, () -> of((Bson) null));
+        // doc inside doc
+        assertExpression(
+                BsonDocument.parse("{'a': {'x': 1, 'y': 2}}"),
+                ofDoc("{a: {x: 1, y: 2}}"),
+                "{'$literal': {'a': {'x': 1, 'y': 2}}}");
+        // empty
+        assertExpression(
+                BsonDocument.parse("{}"),
+                ofDoc("{}"),
+                "{'$literal': {}}");
+        // ensure is literal
+        assertExpression(BsonDocument.parse(
+                "{'lit': {'$not': true}}"),
+                of(BsonDocument.parse("{lit: {'$not': true} }")),
+                "{'$literal': {'lit': {'$not': true}}}");
+    }
+
+    @Test
+    public void getFieldTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/getField/ (100)
+        // these count as assertions by the user that the value is of the correct type
+
+        assertExpression(1, a1.getInteger("a"), "{'$getField': {'input': {'$literal': {'a': 1}}, 'field': 'a'}}");
+        assertExpression(2, a1.getInteger("a").multiply(2), "{'$multiply': [{'$getField': {'input': {'$literal': {'a': 1}}, 'field': 'a'}}, 2]}");
+
+        // different types
+        String getFieldMql = "{'$getField': {'input': {'$literal': {'a': 1}}, 'field': 'a'}}";
+        assertExpression(1, a1.getNumber("a"), getFieldMql);
+        // these are all violations, since they assert the wrong type, but we are testing the generated Mql:
+        assertExpression(1, a1.getBoolean("a"), getFieldMql);
+        assertExpression(1, a1.getInteger("a"), getFieldMql);
+        assertExpression(1, a1.getString("a"), getFieldMql);
+        assertExpression(1, a1.getDate("a"), getFieldMql);
+        assertExpression(1, a1.getArray("a"), getFieldMql);
+        assertExpression(1, a1.getDocument("a"), getFieldMql);
+        // usage with other expressions
+        assertExpression(false, ofDoc("{a: true}").getBoolean("a").not());
+        assertExpression(0.5, ofDoc("{a: 1.0}").getNumber("a").divide(2));
+        assertExpression(8, ofIntegerArray(9, 8, 7).elementAt(ofDoc("{a: 1.0}").getInteger("a")));
+        assertExpression("a", ofDoc("{a: 'A'}").getString("a").toLower());
+        assertExpression(12, ofDoc("{a: {'$date': '2007-12-03T10:15:30.005Z'}}").getDate("a").month());
+        assertExpression(3, ofDoc("{a: [3, 2]}").getArray("a").first());
+        assertExpression(2, ofDoc("{a: {b: 2}}").getDocument("a").getInteger("b"));
+    }
+
+    @Test
+    public void getFieldOrTest() {
+        // getInteger-Or lacks isInteger; isInteger lacks a way to allow doubles that are integers
```

Comment:
There is no `isInteger`. `1.0` is a valid integer (because Integers are a subset of Numbers on the server and therefore in our type system), but there seems to be no clean way to check this. Is there opposition to `i eq i.floor ? true : false` as an `isInteger` check, or perhaps some better alternative?